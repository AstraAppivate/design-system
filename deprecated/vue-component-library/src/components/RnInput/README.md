---
title: Input
description: The NELSON Component Libary provides basic styled inputs and textareas
---

# Basic Example

```html
  <rn-input id="input" type="text" placeholder="Label"></rn-input>
```

#### Props

Prop Name       | Value      | Required
--------------- | ---------- | --------
**id**          | `{String}` | true
**placeholder** | `{String}` | true
**type**        | `{String}` | false

To use a textarea instead of an input, simply leave the `type` blank.
