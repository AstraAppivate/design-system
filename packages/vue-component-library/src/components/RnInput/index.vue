<template lang="html">
  <div class="rn-inputBlock">
    <textarea
      v-if="isTextArea"
      class="textarea"
      :id="id"
      :value="value"
      :rows="rows"
      :cols="cols"
      :name="name"
      @input="updateSelf" ></textarea>
    <input
      v-else
      class="input"
      :class="hasContent ? 'hasContent' : ''"
      :id="id"
      :type="type"
      :value="value"
      :focus="focus"
      :name="name"
      @input="updateSelf"
      @click="setFocus"
      @blur="looseFocus"
    />
    <label class="inputLabel" :for="id">
      <span class="inputLabelInner">{{ label }}</span>
    </label>
  </div>
</template>

<script src="./scripts.js"></script>
