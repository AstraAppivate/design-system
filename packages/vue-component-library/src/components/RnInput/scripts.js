import uuid from 'uuid'

export default {
  name: 'RnInput',
  props: {
    id: {
      type: String,
      default: uuid(),
    },
    name: {
      type: String,
    },
    type: {
      type: String,
      default: 'text',
    },
    label: {
      type: String,
      default: '',
    },
    value: {
      type: String,
      default: '',
    },
    rows: {
      type: Number,
      default: 5,
    },
    cols: {
      type: Number,
      default: 80,
    },
  },
  data: () => ({
    focus: false,
  }),
  computed: {
    isTextArea() {
      return this.type === 'textarea'
    },
    hasContent() {
      return this.value.length || this.focus
    },
  },
  methods: {
    updateSelf(event) {
      this.$emit('input', event.target.value)
    },
    setFocus() {
      this.focus = true
    },
    looseFocus() {
      this.focus = false
    },
  },
}
