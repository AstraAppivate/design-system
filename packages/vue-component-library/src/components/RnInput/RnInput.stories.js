import { storiesOf } from '@storybook/vue'
import RnInput from './index.vue'

storiesOf('Inputs', module)
  .add('Text', () => ({
    components: { RnInput },
    data: () => ({
      user: 'Fred',
    }),
    template:
      '<rn-input type="text" label="User Name" name="user" v-model="user" />',
  }))
  .add('Number', () => ({
    components: { RnInput },
    data: () => ({
      users: '10',
    }),
    template:
      '<rn-input type="number" label="Users" name="user" v-model="users"/>',
  }))
  .add('Textarea', () => ({
    components: { RnInput },
    data: () => ({
      description: '',
    }),
    template:
      '<rn-input type="textarea" label="Description" v-model="description" />',
  }))
