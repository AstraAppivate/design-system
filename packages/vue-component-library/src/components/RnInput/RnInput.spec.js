import { shallow } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import RnInput from '@/src/components/RnInput/index.vue'

describe('RnInput.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(RnInput, {
      propsData: {
        user: 'Fred',
        type: 'text',
        label: 'My Label',
        name: 'user',
        id: '1234',
      },
    })
  })

  it('snapshot: has same HTML structure', () => {
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error(err)
      }

      expect(str).toMatchSnapshot()
    })
  })
})
