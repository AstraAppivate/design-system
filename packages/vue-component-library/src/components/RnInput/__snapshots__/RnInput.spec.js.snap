// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RnInput.vue snapshot: has same HTML structure 1`] = `<div user="Fred" class="rn-inputBlock"><input id="1234" type="text" name="user" value="" class="input"> <label for="1234" class="inputLabel"><span class="inputLabelInner">My Label</span></label></div>`;
